src/ui/DynamicFacet/DynamicFacetValues/DynamicFacetValueCreator.ts

```diff
@@ -12,6 +12,10 @@ export class DynamicFacetValueCreator implements IValueCreator {
     return FacetUtils.getDisplayValueFromValueCaption(value, this.facet.options.field as string, this.facet.options.valueCaption);
   }
 
+  public getDefaultValues() {
```

Comment:
This seems suspicious. Why would a `DynamicFacetValueCreator` (so, something that deals with "single" values) need to return an array here ? With no logic whatsoever, just always empty.

Seems like a patch/hack because the method `getDefaultValues` was added on the `IValueCreator` interface ?

Reply:
I see what you are saying, and yes there can be room to refactor ... but I disagree that the current solution is a patch :) Let me start by the way things are structured at the moment,

The `DynamicFacetValues` class is the entry point here. It acts as an orchestrator and state manager for `DynamicFacet` and `DynamicFacetRange` values. It holds functions like `createFromResponse` and `reset`, but it defers actual creation to a Creator class passed to it via the constuctor.

```
export class DynamicFacetValues {
     constructor(private facet: IDynamicFacet, creatorKlass: IDynamicFacetValueCreatorKlass) {}
}
```

The Creator class is different for the `DynamicFacet` and `DynamicFacetRange`, and holds the differences in behaviour, whereas `DynamicFacetValues` is generic.

On reset, one would assume that the values should be an empty array. However, that's not the case for the Range, for which the request contains partial, pre-built range values that are enriched with counts by the index/searchapi.

Since there is a difference in the default behaviour between the two facets, placing conditional code in the generic `DynamicFacetValues` class would be incorrect. Moreover, the difference is in value creation (or lack there of), so I feel extending the Creator classes is a natural fit.

I guess the core reason as to why the Facet looks strange is really because the FacetRange is strange. An empty Range still has values, whereas a regular Facet is actually empty, hence an empty array.

Reply:
Fair enough 👍